src/Microsoft.Health.Fhir.Core/Features/Search/ConfigureCustomSearchException.cs

```diff
@@ -0,0 +1,33 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search
+{
+    /// <summary>
+    /// The exception that is thrown when the search parameter is not supported.
+    /// </summary>
+    public class ConfigureCustomSearchException : FhirException
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ConfigureCustomSearchException"/> class.
+        /// </summary>
+        /// <param name="error">The error message to include in the operation outcome issues list.</param>
+        public ConfigureCustomSearchException(string error)
+        {
```

Comment:
The pattern we landed on here is that exceptions should be using `Debug.Assert(!string.IsNullOrEmpty(message), "Exception message should not be empty");` to indicate a problem with the message.
I see some other exceptions using EnsureArg, which also isn't great as that will throw another exception from the exception 😕